File: solutions/clojure/rna-transcription/2/src/rna_transcription.clj
(ns rna-transcription
  (:require [clojure.string :as str]))

(defn to-rna [dna]
  (assert (re-find #"^[CATG]+$" dna) "This isn't a DNA strand!")
  (str/join (map {
    "C" "G"
    "G" "C"
    "T" "A"
    "A" "U"
  } (str/split dna #"")))
)

File: solutions/typescript/rna-transcription/2/rna-transcription.ts
export default class Transcriptor {
    static translations = {
        G: 'C',
        C: 'G',
        T: 'A',
        A: 'U',
    }
    toRna(dna: string) {
        return dna.split('').map((value: string) => {
            const nucleotide = Transcriptor.translations[value]
            if (!nucleotide) {
                throw new Error('Invalid input DNA.')
            }
            return nucleotide
        }).join('')
    }
}